src/Build/BackEnd/Components/Communications/NodeProviderOutOfProcBase.cs

```diff
@@ -188,12 +192,15 @@ protected NodeContext GetNode(string msbuildLocation, string commandLineArgs, in
                     _processesToIgnore.Add(nodeLookupKey);
```

Comment:
You could get the same key added multiple times in the bag. Won't affect correctness, but maybe it makes the code more clear if you use `ConcurrentDictionary.GetOrAdd`